index/internal/bcr/indexer/main.go

```diff
@@ -0,0 +1,153 @@
+// Copyright 2025 EngFlow Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"runtime"
+	"sort"
+	"sync"
+
+	"github.com/EngFlow/gazelle_cc/internal/collections"
+
+	"github.com/EngFlow/gazelle_cc/index/internal/bcr"
+	"github.com/EngFlow/gazelle_cc/index/internal/indexer"
+)
+
+// Script responsible for creating index of external header include to the label of extern dependency defining it's rule.
+// It does process all modules defined in https://registry.bazel.build/ and extracts information about defined header for public rules using bazel query.
+// The mapping keys are always in the normalized form of include paths that should be valid when refering using #include directive in C/C++ sources assuming include paths were not overriden
+// The values of the mapping is a string representation on Bazel label where the repository is the name of the module
+// Mappings are always based on the last version of module available in the registry. If the latest available version is yanked then whole module would be skipped.
+//
+// The script needs to checkout (download) sources of each module and execute bazel query using a fresh instance of Bazel server.
+// This step can be ignored if .cache/modules/ contains extracted module informations from previous run.
+//
+// When processing the results of the query script might exclude targets or headers that are assumed to be internal, the excluded files would be written in textual file on the disk.
+// Mapping contains only headers that are assigned to exactly 1 rule. Header with ambigious rule definitions are also written in textual format for manual inspection.
+// It does also use system binaries: git, patch (gpatch is required on MacOs instead to correctly apply patches to Bazel modules) and bazel (bazelisk preferred)
+func main() {
+	cfg := parseFlags()
+
+	bcrClient, err := bcr.CheckoutBazelRegistry(cfg.bcrConfig)
```

Comment:
Nit: since `main` has some difficulty handling errors, in most cases I like to move the body into a function `func run() error`. Then `main` is just:

```
func main() {
  if err := run(); err != nil {
    fmt.Fprintf(os.Stderr, "error: %v\n", err)
    os.Exit(1)
  }
}
```

For more advanced CLIs, you can then add arguments, environment variables, stdout, stderr as parameters to `run`, then call `run` from tests, possibly in parallel.

(Not really needed here, I just think it's a good pattern to follow).